tests/test_federation.py

```diff
@@ -276,3 +276,108 @@ def test_cross_signing_keys_retry(self):
+        )[0]
+        federation_event_handler = self.homeserver.get_federation_event_handler()
+        event2 = make_event_from_dict(
+            {
+                "room_id": self.room_id,
+                "sender": "@baduser:test.serv",
+                "state_key": "@baduser:test.serv",
+                "event_id": "$event2:test.serv",
+                "depth": 1000,
+                "origin_server_ts": 1,
+                "type": "m.room.member",
+                "origin": "test.servx",
+                "auth_events": [],
+                "prev_state": [(most_recent, {})],
+                "prev_events": [(most_recent, {})],
+                "content": {"membership": "join"},
+                "unsigned": {
+                    "malicious garbage": "hackz",
+                    "more warez": "more hackz",
+                    "age": 14,
+                    "invite_room_state": [],
+                },
+            }
+        )
+        self.get_success(
+            federation_event_handler.on_send_membership_event("test.serv", event2)
+        )
+
+        event = self.get_success(self.store.get_event("$event2:test.serv"))
+        event_dict = event.get_dict()
+        self.assertTrue("age" in event_dict["unsigned"])
+        self.assertTrue("hackz" not in event_dict["unsigned"])
+        self.assertTrue("invite_room_state" in event_dict["unsigned"])
+
+    def strip_event_removes_fields_based_on_event_type(self):
+        most_recent = self.get_success(
+            self.homeserver.get_datastore().get_latest_event_ids_in_room(self.room_id)
+        )[0]
+        federation_event_handler = self.homeserver.get_federation_event_handler()
+        event3 = make_event_from_dict(
+            {
+                "room_id": self.room_id,
+                "sender": "@baduser:test.serv",
+                "state_key": "@baduser:test.serv",
+                "event_id": "$event3:test.serv",
+                "depth": 1000,
+                "origin_server_ts": 1,
+                "type": "m.room.power_levels",
+                "origin": "test.servx",
+                "content": {},
+                "auth_events": [],
+                "prev_state": [(most_recent, {})],
+                "prev_events": [(most_recent, {})],
+                "unsigned": {
+                    "malicious garbage": "hackz",
+                    "more warez": "more hackz",
+                    "age": 14,
+                    "invite_room_state": [],
+                },
+            }
+        )
+        self.get_success(federation_event_handler.on_receive_pdu("test.serv", event3))
+
+        event = self.get_success(self.store.get_event("$event3:test.serv"))
+        event_dict = event.get_dict()
+        self.assertTrue("age" in event_dict["unsigned"])
+        self.assertTrue("invite_room_state" not in event_dict["unsigned"])
+        self.assertTrue("more warez" not in event_dict["unsigned"])
```

Comment:
Can use `self.assertIn` and `self.assertNotIn` here for slightly nicer error messages if it fails